src/test/java/com/vmware/vip/i18n/DateFormatUtilOfflineTest.java

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vip.i18n;
+
+import com.vmware.vipclient.i18n.I18nFactory;
+import com.vmware.vipclient.i18n.VIPCfg;
+import com.vmware.vipclient.i18n.base.cache.FormattingCache;
+import com.vmware.vipclient.i18n.base.instances.DateFormatting;
+import com.vmware.vipclient.i18n.exceptions.VIPClientInitException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Date;
+import java.util.Locale;
+
+public class DateFormatUtilOfflineTest extends BaseTestClass {
+
+    final String fullDateForEn = "Monday, November 20, 2017";
+    final String longDateForEn = "November 20, 2017";
+    final String mediumDateForEn = "Nov 20, 2017";
+    final String shortDateForEn = "11/20/17";
+    final String fullTimeForEn = "1:39:24 PM GMT+08:00";
+    final String longTimeForEn = "1:39:24 PM GMT+8";
+    final String mediumTimeForEn = "1:39:24 PM";
+    final String shortTimeForEn = "1:39 PM";
+    final String fullForEn = "Monday, November 20, 2017 at 1:39:24 PM GMT+08:00";
+    final String longForEn = "November 20, 2017 at 1:39:24 PM GMT+8";
+    final String mediumForEn = "Nov 20, 2017, 1:39:24 PM";
+    final String shortForEn = "11/20/17, 1:39 PM";
+
+    final String fullDateForZh = "2017年11月20日星期一";
+    final String longDateForZh = "2017年11月20日";
+    final String mediumDateForZh = "2017年11月20日";
+    final String shortDateForZh = "2017/11/20";
+    final String fullTimeForZh = "GMT+08:00 下午1:39:24";
+    final String longTimeForZh = "GMT+8 下午1:39:24";
+    final String mediumTimeForZh = "下午1:39:24";
+    final String shortTimeForZh = "下午1:39";
+    final String fullForZh = "2017年11月20日星期一 GMT+08:00 下午1:39:24";
+    final String longForZh = "2017年11月20日 GMT+8 下午1:39:24";
+    final String mediumForZh = "2017年11月20日 下午1:39:24";
+    final String shortForZh = "2017/11/20 下午1:39";
+
+    final String fullDateForFr = "lundi 20 novembre 2017";
+    final String longDateForFr = "20 novembre 2017";
+    final String mediumDateForFr = "20 nov. 2017";
+    final String shortDateForFr = "20/11/2017";
+    final String fullTimeForFr = "13:39:24 GMT+08:00";
+    final String longTimeForFr = "13:39:24 GMT+8";
+    final String mediumTimeForFr = "13:39:24";
+    final String shortTimeForFr = "13:39";
+    final String fullForFr = "lundi 20 novembre 2017 à 13:39:24 GMT+08:00";
+    final String longForFr = "20 novembre 2017 à 13:39:24 GMT+8";
+    final String mediumForFr = "20 nov. 2017 à 13:39:24";
+    final String shortForFr = "20/11/2017 13:39";
+
+    final long timestamp = 1511156364801l;
+    final String timeZone = "GMT+8";
+    Date date = new Date(timestamp);
+
+    DateFormatting dateFormatI18n;
+
+    @Before
+    public void init() {
+        VIPCfg gc = VIPCfg.getInstance();
+        try {
+            gc.initialize("vipconfig-offline");
```

Comment:
This will cause other tests to fail because VIPCFG is a singleton. I had added vipconfig-offline before, but Colin and Xiaochao do not want to use it inside the tests because they say that we should not be able to re-initialize VIPCfg. Hence, that .properties file is not used anymore. Only vipconfig.properties is used, and if I have to modify any property for certain unit tests, I do it inside the test itself. See LocaleTest.testGetSupportedLocalesOfflineBundles for an example.

Reply:
As we discussed offline, unit tests is for testing different scenarios, not for mock real scenarios,  re-initialize is acceptable. And use  properties is more convenient, or we need do setMsgOriginsQueue in many places to re-initialize data source.

So I clear data source queue before each initiallization by calling clearDataSource method in the constructor of parent class of all test classes 'BaseTestClass', and problem is resolved.